chevah/trac-to-github
# #2 Trac RST to vanilla RST

Scope
=====

Trac has [their specific flavor of RST](https://trac.edgewall.org/wiki/WikiRestructuredText).

Where there are custom `:trac:` directives, replace them with plain relative links useful in the GitHub-imported wiki. 

In addition, avoid mangling non-RST text. The code should leave [TracWiki formatted](https://trac.edgewall.org/wiki/WikiFormatting) files alone.

(copied part of GitHub ticket #2)

Changes
=======

Replaced dashes with underscores in the Python script name, to be able to test & import. (also updated README, but not sure if scripts are called anywhere else).

Added tests in `test/test_wiki_trac_rst_convert.py`.

Changed `wiki_trac_rst_convert.py`: 
* extended `convert_content` to handle Trac RST `:wiki:` directive.
* filled in `convert_file` to actually do work

What is not yet done:
* Non-RST TracWiki markup does not get converted yet. Will be done in a new ticket: see https://github.com/chevah/trac-to-github/issues/2#issuecomment-764700864
* Manually fix code blocks (before running the script, convert `{{{` and `}}}` to `````). They occur only on 4 (5?) pages.
* Trac tickets are not yet linked, but they are few enough to do manually (just 3 in Development/QA/Server):
```
    :trac:`#738`
```

* Did not yet run tox, isort nor black.
* Must actually run the script and push the results

How to try and test the changes
===============================

reviewers: @adiroiban 

Run tests using `unittest` from the project directory:
`python3.8 -m unittest`

Checking output
-------------------------

I did not find an easy way to verify the GitHub link behavior. The only way to properly test is to run the conversion, then push and check the wiki.

There are some things you can still check locally:

* run the script, and then look at the diff using gitk, with the "Ignore space change" checkbox checked. 
* check what `:trac:` markers are left


The following does not work for GitHub:

> I used the following method to verify the output:
> 
> 1. Get a copy of the wiki repository (here, `server.wiki`). Note: the conversion happens in-place. If you run repeatedly, keep a copy handy.
> 2. Run the conversion script: 
>   `python wiki_trac_rst_convert.py server.wiki`
> 3. Convert a file to HTML: 
>   `rst2html.py server.wiki/Administrative.rst > server.wiki/Administrative.html`
> 4. View the file in the browser: 
>   `firefox server.wiki/Administrative.html`
> 5. Look at the links in the file: if they point to the correct RST file (prompting download), they are correct. If they point to a missing file, they are wrong.
